lib/segment/src/index/field_index/full_text_index/inverted_index.rs

```diff
@@ -56,28 +57,40 @@ impl ParsedQuery {
     }
 }
 
-#[derive(Default)]
-pub struct InvertedIndex {
-    postings: Vec<Option<PostingList>>,
-    pub vocab: HashMap<String, TokenId>,
-    pub point_to_docs: Vec<Option<Document>>,
-    pub points_count: usize,
+pub enum InvertedIndex {
```

Comment:
Defifinion of new Inverted index. Here I aggregate all business logic (estimations, filtering, payload blocks etc). And In mutable/immutable state I store only data and how to access it